src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -48,44 +48,69 @@ public static void generateReposReport(List<RepoConfiguration> configs, String o
         InputStream is = RepoSense.class.getResourceAsStream(TEMPLATE_FILE);
         FileUtil.copyTemplate(is, outputPath);
 
-        for (RepoConfiguration config : configs) {
-            Path repoReportDirectory;
-            try {
-                GitClone.clone(config);
-                repoReportDirectory = Paths.get(outputPath, config.getDisplayName());
-                FileUtil.createDirectory(repoReportDirectory);
-            } catch (GitCloneException gde) {
-                logger.log(Level.WARNING,
-                        "Exception met while trying to clone the repo, will skip this repo.", gde);
-                repoReportDirectory = Paths.get(outputPath, config.getDisplayName());
-                FileUtil.createDirectory(repoReportDirectory);
-                generateEmptyRepoReport(repoReportDirectory.toString());
-                continue;
-            } catch (IOException ioe) {
-                logger.log(Level.WARNING,
-                        "Error has occurred while creating repo directory, will skip this repo.", ioe);
-                continue;
-            } catch (RuntimeException rte) {
-                logger.log(Level.SEVERE, "Error has occurred during analysis, will skip this repo.", rte);
-                continue;
-            }
+        cloneAndAnalyzeRepos(configs, outputPath);
 
-            // preprocess the config and repo
-            updateRepoConfig(config);
-            updateAuthorList(config);
+        try {
+            FileUtil.deleteDirectory(FileUtil.REPOS_ADDRESS);
```

Comment:
instead of cleaning up here, let's make `cleanupDirectories()` a public method and call it at the end of `cloneAndAnalyzeRepos`.

Reply:
Here I'm deleting the entire `repos` folder, but `cleaupDirectories()` is meant for cleaning up and deleting the folder containing the previous repo which has been analyzed.

Reply:
As in, we should have a clean up method afterward to ensure all the folders are deleted.

Reply:
Not sure I understand. All repos are cloned into the `repos` folder, so deleting the `repos` folder would ensure all the folders are deleted.